src/workers/search/get-matches.js

```diff
@@ -20,8 +20,17 @@ export default function getMatches(
   for (let i = 0; i < lines.length; i++) {
     let singleMatch;
     const line = lines[i];
+    let previousLastIndex = 0;
     while ((singleMatch = regexQuery.exec(line)) !== null) {
       matchedLocations.push({ line: i, ch: singleMatch.index });
+
+      // exec() is supposed to update the lastIndex property but sometimes
+      // doesn't and can cause this to be an infinite loop by matching on the
+      // same index.  A work around is to update the lastIndex manually
+      if (regexQuery.lastIndex === previousLastIndex) {
+        regexQuery.lastIndex++;
+      }
```

Comment:
After looking over this again I didn't like it as much as it seemed to hacky.  But I when I looked up [lastIndex on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex) it seems like setting `lastIndex` yourself to indicate the next index to start matching at is something there by design and is in the spec.  So maybe this is OK after all.